### PR TITLE
feat(nodes): add tool_calendar — Google Calendar tool node

### LINKED ISSUE #1059: feat(nodes): tool_calendar — Google Calendar tool node
Part of the GSuite tool integrations epic. Depends on the shared resolver issue (core/google_access.py).

## Problem Statement

Agents cannot read or manage calendars and events. We need a `tool_calendar` node exposing the full Calendar surface, with deletion gated.

## Proposed Solution

Build `nodes/src/nodes/tool_calendar/` following the `tool_github` reference pattern. Build on the Calendar API v3 via `google-api-python-client`.

### Node anatomy

Standard node anatomy and deps (see epic). `IGlobal.beginGlobal()` calls `resolve_google_access(cfg, CALENDAR)`. Auth: merge the shared `google.*` block (document both paths).

### Full tool surface (no omissions)

Reads (Tier 1): event_list (accept and return syncToken for incremental sync), event_get, event_instances (expand a recurring series), freebusy_query, calendar_list, calendar_get, acl_list.
Write tier (Tier 2): event_create, event_update, event_move (between calendars), event_quick_add (natural-language create), calendar_create, calendar_update, acl_insert, acl_delete (calendar sharing). Attendee-bearing create and update send invites at write tier; this is expected behavior, not gated.
Gated by `allowDelete` (Tier 3): event_delete, calendar_delete.

### Access enum and scopes

`access`: readonly, write. Resolved by the CALENDAR spec (readonly -> calendar.readonly; write -> calendar, required for ACL operations). Flag: allowDelete.

## Acceptance Criteria / Definition of Done

- Every function above implemented as a registered `@tool_function`; event_delete and calendar_delete error clearly when allowDelete is off.
- event_list accepts and returns syncToken; documented for incremental sync.
- event_instances expands a recurring series; event_move targets a destination calendarId.
- Attendee invites: documented that create and update with attendees send invitations at write tier (sendUpdates handled per API), not gated.
- Input schemas list required vs optional fields with defaults; outputs cleaned (id, status, start, end, attendees, htmlLink), not raw JSON. calendarId and eventId are invoke-time params.
- Tests: success and error per function, mock run under `ROCKETRIDE_MOCK`, and a test asserting the CALENDAR flag name (`allowDelete`) exists as a `services.json` field.

## Config panel

Merges the shared Google auth block, plus:

```json
"fields": [
  { "name": "access", "type": "select", "default": "write",
    "options": ["readonly", "write"], "label": "Access level" },
  { "name": "allowDelete", "type": "boolean", "default": false,
    "label": "Allow event / calendar deletion (irreversible)" }
]
```

## Alternatives Considered

Single combined `tool_gsuite` node: rejected (tool-surface size, per-app access).

## Affected Modules

- [x] nodes (pipeline)
- [x] ai

Part of #1053.

### PR DESCRIPTION
**Pull request by kgarg2468**
## Summary

- Add `tool_calendar` — an agent-invocable Google Calendar tool node (mirrors the shipped `tool_gmail`: shared `core/google_access.py` auth, user-OAuth + service-account).
- 18 operations across events (`event_list`/`event_get`/`event_instances`/`event_create`/`event_update`/`event_move`/`event_quick_add`/`event_delete`), calendars (`calendar_list`/`calendar_get`/`calendar_create`/`calendar_update`/`calendar_delete`), ACLs (`acl_list`/`acl_insert`/`acl_delete`), `freebusy_query`, and `check_connection`.
- Safety gate: `event_delete` and `calendar_delete` require the `allowDelete` flag (`acl_delete` is deliberately write-tier only). `sendUpdates` defaults to `'all'` and is always sent explicitly so attendee notifications aren't silently suppressed.
- Includes a contract fix (commit `c1b5996f`): the event cleaner now preserves `description` and `location` on every event-returning operation (previously dropped), and the returned-field contract is stated in the event tool descriptions so agents know the metadata is available.

## Type

feature (new agent-tool node) + tests

## Testing

- [x] Tests added or updated — 50 unit + service/contract tests (`nodes/test/tool_calendar/`), incl. the writable-metadata regression
- [x] Tested locally — Python 3.12: 50 unit pass, 294 contract tests pass, `ruff check`/`format` clean
- [ ] `./builder test` passes — not run locally (full C++ engine recompile); node contract/dynamic tests run in CI

Live end-to-end as a real user (user-OAuth token, personal GCP project): all 18 operations against the real Calendar API, plus both gate directions (`event_delete`/`calendar_delete` denied without `allowDelete`, succeed with it) and the ACL insert→revoke lifecycle. `description`/`location` round-trip verified live and through an `agent_rocketride` canvas run.

## Checklist

- [x] Commit messages follow [conventional commits](https://www.conventionalcommits.org/)
- [x] No secrets or credentials included
- [ ] Wiki updated (if applicable) — n/a; node README with generated marker block included
- [x] Breaking changes documented (if applicable) — none (purely additive node)

## Linked Issue

Closes #1059

## Summary by CodeRabbit

* **New Features**
  * Added Google Calendar tool support (events, calendars, ACL rules, free/busy, recurring expansion) with connection diagnostics.
  * Introduced shared global setup for authenticated access, supporting service accounts and user OAuth with scope-aware authorization.
  * Added configurable deletion controls via `calendar.allowDelete`, with clearable update fields.
* **Documentation**
  * Added detailed tool documentation covering auth modes, parameters, limits, and troubleshooting.
* **Bug Fixes / Improvements**
  * Improved request robustness with retries and normalized agent-friendly outputs.
* **Tests**
  * Expanded unit tests for validation, output shaping, deletion gating, token/scope checks, and retry/rate-limit behavior.

### CODE REVIEW

File: nodes/src/nodes/tool_calendar/calendar_client.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+                continue
+            detail = getattr(exc, 'reason', None) or str(exc)
+            if status and int(status) == 403:
+                raise ValueError(
+                    f'Google Calendar API 403: {detail}. If this is a scope error, disconnect '
+                    'and reconnect your Google account with the required access tier.'
+                ) from exc
+            prefix = f'Google Calendar API {status}: ' if status else 'Google Calendar request failed: '
+            raise ValueError(f'{prefix}{detail}') from exc
+    raise RuntimeError('execute: retry loop exhausted unexpectedly')  # unreachable
```
[the remark this answers is not in the document]
Comment (kgarg2468) [reply nodes/src/nodes/tool_calendar/calendar_client.py]: Partially adopted, deliberately: the 403-by-reason half is implemented — rate/quota-limit 403s back off and retry while scope/permission 403s fail fast (f40f79c9), now parsing the structured `error.errors[].reason` body (bdf13f7b). The mutation-idempotency half is intentionally unchanged: a 403/429 means Google rejected the request (nothing was applied), and the 429/5xx retry shape mirrors the merged `tool_gmail` client this node clones verbatim — every Google Workspace node in the family shares it. Making write retries idempotency-aware is a family-wide change to the shared client pattern, out of scope for this node PR; happy to file a follow-up issue if maintainers want it.

File: nodes/src/nodes/tool_google_workspace/calendar/IInstance.py [hunk 2]
```diff
@@ -0,0 +1,830 @@
...
+    def _opt_str_list(args: dict, key: str, op: str) -> list[str] | None:
+        """Validate an optional non-empty list of non-empty strings (e.g. RRULE recurrence lines)."""
+        value = args.get(key)
+        if value is None:
+            return None
+        if not isinstance(value, list) or not value:
+            raise ValueError(f'{op}: "{key}" must be a non-empty list')
+        if not all(isinstance(i, str) and i.strip() for i in value):
+            raise ValueError(f'{op}: "{key}" must contain only non-empty strings')
+        return value
```
[the remark this answers is not in the document]
Comment (kgarg2468) [reply nodes/src/nodes/tool_google_workspace/calendar/IInstance.py:174]: Addressed: `event_update` accepts empty strings to clear `description`/`location` via `_opt_clearable_str` (f40f79c9), and `calendar_update.description` is now clearable the same way with a body-level test (bdf13f7b). `recurrence` is deliberately not clearable here: list-typed patch fields have different clearing semantics (empty-array vs null varies by field) and silently converting a recurring event to a single event is a sharper edge than clearing a string — flagging for a follow-up rather than bundling it.

File: nodes/src/nodes/tool_calendar/IInstance.py [hunk 3]
```diff
@@ -0,0 +1,830 @@
...
+
+                    info = _json.loads(_decode_blob(token_str))
+                    granted = {s for s in (info.get('scope') or '').split() if s}
+                    covered = _scopes_satisfied(granted, out['requiredScopes'])
+                    out['connection_ok'] = out['connection_ok'] and covered
+                    if not covered:
+                        out['missingScopes'] = [s for s in out['requiredScopes'] if s not in granted]
+        except Exception:
+            pass  # diagnostics must never raise
+        return out
```
[the remark this answers is not in the document]
Comment (kgarg2468) [reply nodes/src/nodes/tool_calendar/IInstance.py]: Done in f40f79c9: `check_connection` now issues a live `calendarList().list(maxResults=1)` probe when a service handle is present; a probe failure reports `connection_ok=False` plus a truncated `error`. The scope-coverage diagnostics remain, and the probe is guarded so stubbed/contract environments are unaffected.

File: nodes/src/nodes/tool_google_workspace/google_client.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+                        ) from exc
+                    except (ValueError, json.JSONDecodeError) as exc:
+                        raise _gae.RefreshError(
+                            'Google Calendar token refresh returned a malformed response. '
+                            'Please reconnect your Google account.'
+                        ) from exc
+                    self.token = data.get('access_token') or self.token
+                    ms = data.get('expiry_date')
+                    if ms:
+                        self.expiry = _dt.datetime.fromtimestamp(ms / 1000, tz=_dt.timezone.utc).replace(tzinfo=None)
```
[the remark this answers is not in the document]
Comment (kgarg2468) [reply nodes/src/nodes/tool_google_workspace/google_client.py]: Both fixed in bdf13f7b: (1) `HTTPError` is caught before `URLError`, so a broker rejection reports "rejected by the broker (HTTP <code>)" instead of "unreachable"; (2) a 200 response without `access_token` raises `RefreshError` instead of silently reusing the stale token. SDK-gated tests cover both paths.

File: nodes/test/tool_calendar/test_calendar.py [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+        ('event_move', {'eventId': 'ev1', 'destination': 'other'}, 'move', _EVENT),
+    ],
+)
+def test_all_event_returning_ops_preserve_cleaned_description_and_location(op, args, result_key, result):
+    inst = _make(results={result_key: result})
+    out = getattr(inst, op)(args)
+    event = out['events'][0] if 'events' in out else out
+    assert event['description'] == _EVENT['description']
+    assert event['location'] == _EVENT['location']
+
```
[the remark this answers is not in the document]
Comment (kgarg2468) [reply nodes/test/tool_calendar/test_calendar.py]: Added in bdf13f7b: `event_quick_add` is now in the parametrized cleaned-field test (`quickAdd` result key), asserting the `description`/`location` preservation contract across all 7 event-returning operations.

File: nodes/src/nodes/tool_calendar/calendar_client.py [hunk 6]
```diff
@@ -0,0 +1,478 @@
...
+    """
+    token_str = str(cfg.get('userToken') or '').strip()
+    if not token_str:
+        return set(), True, []
+    info = json.loads(_decode_blob(token_str))
+    granted = {s for s in (info.get('scope') or '').split() if s}
+    if not granted or _scopes_satisfied(granted, required):
+        return granted, True, []
+    return granted, False, [s for s in required if s not in granted]
+
```
[the remark this answers is not in the document]
Comment (kgarg2468) [reply nodes/src/nodes/tool_calendar/calendar_client.py]: Done — `build_service` now routes its scope check through `token_scope_report` too (drive already did via `_scope_coverage`), so the last independent copy is gone. Applied to sheets/docs/calendar in the latest commits.

**Review by asclearuc — CHANGES_REQUESTED** [review]
Directories:
- `tool_drive` -> `tool_google_drive`
- `tool_calendar` -> `tool_google_calendar`
- `tool_sheets` -> `tool_google_sheets`
- `tool_docs` -> `tool_google_docs` 

Rationale:
- together in the directory list
- if would be another calendar (from MS for example,) it would be clear which one which

File: nodes/src/nodes/tool_google_workspace/calendar/IInstance.py [hunk 7]
```diff
@@ -0,0 +1,863 @@
...
+    )
+    def acl_insert(self, args: dict) -> dict:
+        """Grant a calendar ACL rule (share the calendar). Requires the write tier."""
+        args = normalize_tool_input(args, tool_name='tool_calendar')
+        self._access().require_write('acl_insert')
+        cal_id = self._calendar_id(args)
+        role = self._enum_arg(args, 'role', _ACL_ROLES, None)
+        if role is None:
+            raise ValueError(f'acl_insert: "role" is required and must be one of {list(_ACL_ROLES)}')
+        scope_type = self._enum_arg(args, 'scopeType', _ACL_SCOPE_TYPES, None)
```
Comment (asclearuc) [inline nodes/src/nodes/tool_google_workspace/calendar/IInstance.py:736]: **medium (security)** — `acl_insert` can make a calendar public with no extra gate.

With `scopeType: "default"` the calendar becomes readable by anyone on the internet; with `"domain"`, by the whole company — and both only need the write tier. The PR gates *deletion* behind `allowDelete`, but *exposure* is at least as dangerous for an agent tool: a prompt-injected agent (e.g. via a processed document saying "share the calendar publicly") can leak every event silently. Deleted events are gone; leaked events are copied.

The codebase already has the exact mechanism: the `DRIVE` spec in `nodes/core/google_access.py` declares an `allowPublicSharing` flag. Suggestion:

1. add `'allowPublicSharing'` to the `CALENDAR` spec's `flags`,
2. here, call `self._access().require_flag('allowPublicSharing', 'acl_insert')` when `scope_type` is `'default'` (please consider `'domain'` too),
3. keep normal `user`/`group` sharing at plain write tier.

If you believe calendar sharing should stay ungated, that's a fair position — but then please state the reasoning in the README's safety section so it reads as a decision, not an omission.

File: nodes/src/nodes/tool_calendar/calendar_client.py [hunk 8]
```diff
@@ -0,0 +1,477 @@
...
+"""
+Google Calendar API v3 client helpers.
+
+Credential construction (service account or user OAuth), the discovery-built
+service, and response cleaners that turn raw Calendar JSON into compact,
+agent-friendly shapes. All tool methods in IInstance call through here.
+
+The credential machinery (service-account + broker-backed user OAuth) is shared
+verbatim with the other Google Workspace tool nodes; keep it in step with
```
Comment (asclearuc) [inline nodes/src/nodes/tool_calendar/calendar_client.py]: **low (maintainability / security follow-up)** — this copy already diverged from `tool_gmail`, and gmail is left with the unfixed bugs.

"Shared verbatim" is not accurate: this copy adds `resolve_token_uri`, `_is_rate_limit_403` retry handling, the schemeless `RR_OAUTH_BROKER_URL` fallback, and `token_scope_report` — none of which exist in gmail. A comment is not a sync mechanism. All good fixes — but `gmail_client.py:139` still trusts `token_uri` straight from the saved token, which is exactly the tampered-token risk this file's own comments describe.

Two asks:
- Please open a tracking issue to backport at least the `token_uri` validation to `tool_gmail` (follow-up PR is fine).
- Longer term, move this machinery to a shared module next to `nodes/core/google_access.py` (e.g. `nodes/core/google_client.py`) so a third Google node does not create a third copy.

File: nodes/src/nodes/tool_google_workspace/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+google-api-python-client
+google-auth
+google-auth-oauthlib
+google-auth-httplib2
+# Transitive floors: OSV flagged the resolved versions (CVE-2026-45409 idna,
+# CVE-2025-4565 protobuf).
+idna>=3.15
+protobuf>=5.29.6
```
Comment (asclearuc) [inline nodes/src/nodes/tool_google_workspace/requirements.txt:8]: **nit** — calendar pins `protobuf>=5.29.6` while gmail pins `>=4.25.8`. Both satisfy CVE-2025-4565, so this works — but if the two nodes ever resolve into one shared environment, aligned floors avoid confusion. Intentional?

**Review by asclearuc — COMMENTED** [review]
Thanks @kgarg2468 — each of these five Google nodes (4 news and already existing `gmail`) is individually well built: the access-tier gating, the broker-refresh hardening, and the test coverage are all solid. Before we merge them, though, I'd like to request one structural change, and it applies across the whole set — #1549 (sheets), #1550 (docs), #1551 (calendar), #1552 (drive), and the already-shipped `tool_gmail`.

## Request: aggregate all five under one `tool_google_workspace` package

Right now each node is a standalone package, and each ships its own copy of the same ~200-line credential/client machinery (`build_service`, `_BrokerCredentials`, `resolve_refresh_url`, `resolve_token_uri`, `token_scope_report`, `_decode_blob`, `execute`, `_is_rate_limit_403`). That is already causing real drift: the newer nodes validate `token_uri` against the trusted Google hosts, but `gmail_client.py` still passes a token-supplied `token_uri` through unvalidated. Code that guards refresh tokens should not live in five hand-synced copies.

**This platform already supports the layout I'm asking for — `agent_crewai` ships it today.** It has:
- top-level shared libraries (`crewai_base.py`, `crewai_runner.py`, `crewai_listener.py`) + one shared `requirements.txt` + one icon,
- several `services.*.json` files in one directory (`services.agent.json`, `services.manager.json`, `services.subagent.json`),
- one per-service subpackage each with its own `IGlobal.py` / `IInstance.py` (`crewai_agent/`, `crewai_manager/`, `crewai_subagent/`),
- a shared base class subclassed by each service.

Please follow the same pattern for the Google nodes:

```
nodes/src/nodes/tool_google_workspace/
  __init__.py
  requirements.txt
  google.svg
  google_client.py        # the shared credential/client machinery (one copy)
  IGlobal.py              # base lifecycle: beginGlobal / validateConfig / endGlobal
  IInstance.py            # base helpers: _svc(), _access(), check_connection, arg validators
  services.gmail.json     # path: nodes.tool_google_workspace.gmail
  services.sheets.json
  services.docs.json
  services.calendar.json
  services.drive.json
  gmail/    IGlobal.py IInstance.py README.md __init__.py   # tool functions + cleaners only
  sheets/   ...
  docs/     ...
  calendar/ ...
  drive/    ...
```

What moves to the shared base: the credential machinery, the `beginGlobal`/`validateConfig`/`endGlobal` lifecycle (parameterized by the `AccessSpec` + API name/version), `check_connection`, and the int/number arg validators. The `AccessSpec` definitions already live shared in `nodes/core/google_access.py` — reuse them as-is. What stays in each service subpackage: only its `@tool_function` methods, its response cleaners, and its field schema in `services.<svc>.json`.

## Scope / what does not change

This consolidates code, not the tools. It stays five separate tools with five `services.*.json` files. Keep each node's existing `protocol` (`tool_gmail://`, `tool_sheets://`, …) and `prefix` unchanged, so saved `.pipe` files — which reference the protocol, not the Python path — keep working. Only the internal `path` changes (e.g. `nodes.tool_gmail` → `nodes.tool_google_workspace.gmail`), exactly like `agent_crewai`'s `services.*.json` point `path` at `nodes.agent_crewai.crewai_agent`. Icons, tiles, and access tiers are unchanged. (This also supersedes the earlier `tool_drive` → `tool_google_drive` rename request — under the umbrella it becomes `tool_google_workspace/drive`.)

## Suggested sequencing (so your work is not thrown away)

The service-specific code you already wrote — tool functions, cleaners, tests — transfers almost unchanged; only the `IGlobal`/client boilerplate collapses into the base. I'd suggest:
1. First PR: move the shipped `tool_gmail` into the new umbrella and add the shared `google_client.py` + base classes. This is the template and it fixes the `token_uri` security drift immediately.
2. Then rebase #1549 / #1550 / #1551 / #1552 onto the umbrella so each becomes a thin service subpackage.

Happy to pair on the base-class shape if useful. Requesting changes so we settle the structure before these four land as five separate packages.

File: nodes/src/nodes/tool_google_workspace/calendar/IInstance.py (hunk 7 above)
Comment (kgarg2468) [reply nodes/src/nodes/tool_google_workspace/calendar/IInstance.py:736]: Fixed in a95c51d0 — acl_insert with scopeType `default` or `domain` now requires a new `allowPublicSharing` flag (off by default), mirroring tool_drive's sharing gate; individual user/group grants stay write-tier. Your framing ("leaked events can't be un-leaked") is in the code comment. Note this adds one line to the shared `nodes/core/google_access.py` (the CALENDAR AccessSpec flags tuple) — called out in the commit message per the shared-code rule. Gate is documented in services.json, the README table, and the tool description; deny/allow paths tested.

File: nodes/src/nodes/tool_calendar/calendar_client.py (hunk 8 above)
Comment (kgarg2468) [reply nodes/src/nodes/tool_calendar/calendar_client.py]: Fixed in a95c51d0 — docstring corrected; gmail backport (incl. the token_uri trust gap you cite) + nodes/core hoist tracked in #1560.

File: nodes/src/nodes/tool_google_workspace/requirements.txt (hunk 9 above)
Comment (kgarg2468) [reply nodes/src/nodes/tool_google_workspace/requirements.txt:8]: Semi-intentional: the four new Workspace nodes took the current-secure floor (>=5.29.6, per CodeRabbit's CVE pass) while gmail predates it. Aligning gmail's floor is folded into the gmail-backport follow-up #1560 rather than this PR, so the five Google nodes converge in one place. Happy to bump it here instead if you'd rather not wait.

**Review by joshuadarron — CHANGES_REQUESTED** [review]
Reviewed the full diff (10 files, +2635). The node itself is in good shape — I found no correctness or security bugs in the new code, and CI is green. Requesting changes on structure only, in line with @asclearuc's pending review.

## Blocking

**`calendar_client.py` ships another full copy (~300 lines) of the credential/refresh machinery** (`build_service`, `_BrokerCredentials`, `resolve_refresh_url`, `resolve_token_uri`, `execute`, `_is_rate_limit_403`). This is the 2nd in-tree copy (5th counting #1549/#1550/#1552), and drift is already real: this copy validates `token_uri` against trusted Google hosts, while `tool_gmail`'s copy still passes a token-supplied `token_uri` through unvalidated. This code guards refresh tokens — hand-synced copies of it are exactly how the gmail gap happened. Our architecture rules say extract on the second consumer: please land the shared client first (`nodes/core/google_client.py`, or the `tool_google_workspace` base from @asclearuc's proposal — either works for me), then rebase this PR into a thin service layer on top. #1560 tracks the hoist, but a tracking issue shouldn't wave a 5th copy in; sequencing the shared client first fixes the gmail `token_uri` gap immediately too.

## Non-blocking findings

- `calendar_client.py:55,171`: 🔵 `_SUPERSET_SCOPES` / `_scopes_satisfied` reimplement `nodes/core/google_access.py::_scope_satisfied` / `missing_scopes`, which are already superset-aware for calendar (write implies readonly). `gmail_client.py` imports `missing_scopes` from core — `token_scope_report` should too.
- `requirements.txt:8`: 🔵 protobuf floor `>=5.29.6` vs gmail's `>=4.25.8`. Align the floors (asclearuc's open nit).
- `calendar_client.py:357`: 🔵 the `min(..., 60.0)` cap is dead code — with 4 attempts the max sleep is 4s. Drop the cap or note the real ceiling; as written it implies waits that can't happen.

## Good

- Safety gating is exactly right: `allowDelete` for event/calendar deletion, `allowPublicSharing` for `default`/`domain` ACL rules (cleanly addresses the earlier exposure finding), `sendUpdates` always sent explicitly so Google's implicit `none` can't silently suppress notifications, and the fail-loud non-bool flag resolver.
- The broker/token hardening (https + host allowlist for both the refresh URL and `token_uri`) is a real improvement over the gmail baseline.
- Test coverage is thorough: both gate directions, readonly denial across all 10 write ops, maxResults clamping and bool rejection, syncToken/filter conflicts, and the broker refresh edge cases (missing token, HTTP rejection, stale expiry, read timeout).

Once the shared-client hoist lands and this rebases onto it, I'm happy to approve — the tool-function layer, cleaners, and tests here transfer essentially as-is.

**Comment by kgarg2468** [comment]
Shared-client extraction is up: #1570 adds `tool_google_workspace/` (one hardened credential machinery, `agent_crewai`-style multi-service layout) and migrates `tool_gmail` onto it — closing the gmail `token_uri` gap immediately, as requested. Once #1570 lands, this PR rebases into its thin service subpackage; the rebase is staged locally and will be pushed after #1570 merges (or earlier as a stacked preview if you prefer). cc @joshuadarron @asclearuc

**Comment by kgarg2468** [comment]
Restructured as requested: this PR is now a **thin service layer on the shared client** — one commit adding only the service subpackage (tool functions + response cleaners), its `services.<svc>.json` (`path`-only change; protocol/prefix/fields untouched), icon, and tests. It is stacked on #1570, so the diff currently shows the two umbrella commits too; once #1570 merges this rebases to just the service commit. All previous review fixes are preserved (they live in the shared `google_client.py` in #1570, or in the service layer here). Tests pass with and without the Google SDK installed; 296 contract tests green.
